tests/unit/amo/pages/TestAddonReviewList.js

```diff
@@ -149,139 +164,99 @@ describe(__filename, () => {
 
   describe('<AddonReviewList/>', () => {
     it('shows placeholders for reviews without an addon', () => {
-      const root = render({ addon: null });
+      render();
 
+      const cards = screen.getAllByClassName('AddonReviewCard');
       // Make sure four review placeholders were rendered.
-      expect(root.find(AddonReviewCard)).toHaveLength(4);
-      // Do a sanity check on the first placeholder;
-      expect(root.find(AddonReviewCard).at(0)).toHaveProp('addon', null);
-      expect(root.find(AddonReviewCard).at(0)).toHaveProp('review', null);
+      expect(cards).toHaveLength(4);
+      cards.forEach((card) =>
+        expect(within(card).getAllByRole('alert')).toHaveLength(2),
+      );
     });
 
     it('displays 4 placeholders while loading for an addon with no reviews', () => {
-      const externalAddon = {
-        ...fakeAddon,
-        ratings: {
-          ...fakeAddon.ratings,
-          count: 0,
-        },
-      };
-      const addon = createInternalAddonWithLang(externalAddon);
-
-      _loadAddon(externalAddon);
-
-      const root = render();
+      addon.ratings = { ...addon.ratings, count: 0 };
```

Comment:
I don't understand the question. This test is for an add-on with zero ratings, while the next test is for an add-on with 6 ratings. Each tests loads an add-on first, but doesn't load reviews, and it's this second fact that causes the placeholders for ratings to appear.

Reply:
I was reading them and this test does:
```
      addon.ratings = { ...addon.ratings, count: 0 };
      renderWithAddon();
      const cards = screen.getAllByClassName('AddonReviewCard');
      expect(cards).toHaveLength(4)
```
and it's testing that there are 4 placeholders while loading with there being 0 count ratings.

Wheres the following is finished loading because there are 6 placeholders for 6 ratings?
```
      addon.ratings = { ...addon.ratings, count: 6 };
      renderWithAddon();
      const cards = screen.getAllByClassName('AddonReviewCard');
      expect(cards).toHaveLength(6)
```

I can't see anything different in the setup that would mark the add-on as loading or force it to load.  Or are they both testing while loading and we only render the correct number of placeholders if the count > 0? (why?)

Reply:
There are a few things at play:
1. What is loading are the reviews, not the add-on. In both tests the add-on is already loaded.
2. If the add-on has 0 reviews, then there is code that renders 4 review placeholders.
3. If the add-on has > 0 reviews, then we use the number of reviews for the number of placeholder.

That's why the first test expects 4 placeholders (because `count` is 0), and why the second test expects 6 placeholders (because `count` is 6).

Reply:
Okay.  Do you know the history behind the decision to load 4 placeholders when the add-on has 0 reviews? (but to load the correct count of placeholders for >0).  Or why we bother trying to get the reviews if we already know the count is 0?

Reply:
Looking at https://github.com/mozilla/addons-frontend/issues/2664 and https://github.com/mozilla/addons-frontend/pull/7056, it seems like the intent is to use 4 loading indicators when we don't know the count, so I think you are correct that we should probably have different logic when we know there are exactly 0 reviews (i.e., show 0 placeholders). It probably makes sense to file a separate issue for this. I can do so.